### PR TITLE
tty: notify driver when there's space in the input queue.

### PR DESCRIPTION
**Pull request by j-piecuch**
This PR kicks off a series of (hopefully small) PRs with an end goal of implementing pseudoterminals.
Why we need this PR:
`write()` calls on a PTY master device can block due to the slave device's input queue being full. We need some means of notifying the master that space has become available.

### CODE REVIEW

**Review by wiklam — APPROVED** [review]
LGTM

File: sys/kern/tty.c [hunk 1]
```diff
@@ -493,17 +495,28 @@ void tty_input(tty_t *tty, uint8_t c) {
...
+      return false;
     }
 
     ringbuf_putb(&tty->t_inq, c);
     tty_wakeup(tty);
+    return true;
+  }
+}
+
+static void tty_check_in_lowat(tty_t *tty) {
```
Comment (cahirwpz) [inline sys/kern/tty.c:512]: Could you adda comment that `&tty->t_lock` should be held ?

File: sys/kern/tty.c [hunk 2]
```diff
@@ -493,17 +495,28 @@ void tty_input(tty_t *tty, uint8_t c) {
...
+  if (!(tty->t_flags & TF_IN_HIWAT))
     return;
+
+  if (tty->t_inq.count < TTY_IN_LOW_WATER) {
+    tty->t_flags &= ~TF_IN_HIWAT;
+    if (tty->t_ops.t_notify_in)
```
Comment (cahirwpz) [inline sys/kern/tty.c:518]: Consider introducing `tty` method wrapper as mentioned in the other review.

**Review by cahirwpz — APPROVED** [review]
LGTM though two small improvements could be added.

File: sys/kern/tty.c (hunk 2 above)
Comment (j-piecuch) [reply sys/kern/tty.c]: Done

File: sys/kern/tty.c (hunk 1 above)
Comment (j-piecuch) [reply sys/kern/tty.c:519]: I have added an assertion instead.

**Pull request merged**